reference/auth.v1.yaml

```diff
@@ -680,6 +680,159 @@ paths:
+      tags: []
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: ./auth/models/consent/consents.v1.yaml
+      operationId: ListConsents
+      description: Retrieves the list of all different consent types and their content. By default returns only the latest version for each consent type. This can be override by setting the `version` query parameter to `all`
+      requestBody:
+        content: {}
+      parameters:
+        - schema:
+            type: string
+            enum:
+              - latest
+              - all
+            default: latest
+          in: query
+          name: version
+          description: Whether to return all versions or only the most recent one.
+    parameters: []
+  '/v1/consents/{type}':
+    parameters:
+      - schema:
+          type: string
+        name: type
+        in: path
+        required: true
+    get:
+      summary: Get Latest Consent By Type
+      tags: []
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: ./auth/models/consent/consent.v1.yaml
+      operationId: GetLatestConsentByType
+      description: Returns the latest consent version for the given type
+  '/v1/consent/{type}/versions':
+    parameters:
+      - schema:
+          type: string
+        name: type
+        in: path
+        required: true
+    get:
+      summary: Get Consent Versions
+      tags: []
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: ./auth/models/consent/consents.v1.yaml
+      operationId: GetConentVersions
+      description: Returns a list of all consent versions for a given type
+  '/v1/users/{userId}/consents':
+    parameters:
+      - schema:
+          type: string
+        name: userId
+        in: path
+        required: true
+    get:
+      summary: Get User Consent Records
+      tags: []
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: ./auth/models/consent/records.v1.yaml
+      operationId: GetUserConsentRecords
+      description: Returns a list of all user consent records. By default only the most recent consent record for a given consent type will be returned. This can be overriden by setting the `version` query parameter to `all`.
+      parameters:
+        - schema:
+            type: string
+            enum:
+              - latest
+              - all
+            default: latest
+          in: query
+          name: version
+          description: Whether to return all versions or only the most recent one.
+        - schema:
+            type: string
+          in: query
+          name: type
+    post:
+      summary: Create User Consent Record
```

Comment:
Can a user grant consent only the latest version of a consent? What happens if they try to grant consent to older version?
Can a user grant consent to something they already consented to (i.e. duplicate record)?

This may need a bit more description of the various workflows - are these correct interpretations of the proposed API?
1. Granting consent for the first time: `CreateUserConsentRecord` with latest version
2. Updating a consent to agree to a new version: `CreateUserConsentRecord` with a newer latest version
3. Updating a consent with new metadata: `UpdateUserConsentRecord` with new metadata _only_
4. Updating a consent to revoke it: `UpdateUserConsentRecord` with `status` = `revoked` _only_

Reply:
1. Yes
2. Yes, but it revokes previous consents for the same type
3. Need to think some more about the metadata
4. Yes

Reply:
Added `DELETE /v1/users/{userId}/consents/{recordId}` for 4. (revoking a consent record)

Reply:
How we interpret revocation dates and whether we purge the data depends on the type of the consent and the actual content. Currently, I believe we state that date shared with BDDP accounts is not purged even after the consent is revoked.